File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    var answer = 0;
    const map = {};
    
    for(let i=0; i<priorities.length; i++){
        map[i] = priorities[i];
    }
    
    const processes = Object.entries(map);
    
    while(processes.length!==0){
        const first = processes.shift();
        
        const flag = processes.filter(p=>p[1]>first[1]);
        if(flag.length!==0){
            processes.push(first);
        }
        else{
            answer++;
            if(Number(first[0])===location){
                return answer;
            }
        }
    }
    
    
    return answer;
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    queue = deque()
    
    for i, p in enumerate(priorities):
        queue.append((i, p))
    
    cnt = 0
    while queue:
        idx, p = queue.popleft()
        
        flag = False
        for idx_2, p_2 in queue:
            if p_2 > p:
                flag = True
                queue.append((idx, p))
                break
        
        if flag == False:
            cnt += 1
            if idx == location:
                return cnt
        
        
    
    return 0